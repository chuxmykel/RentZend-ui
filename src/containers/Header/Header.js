import React from 'react';
import { AppBar, Toolbar, Typography } from '@material-ui/core';
import { Link } from 'react-router-dom';

const Header = () => {
  return (
    <AppBar position="static">
      <Toolbar>
        <div className="header">
          <Typography variant="h6">
            <Link to="/">View Agents</Link>
          </Typography>
          <Typography variant="h6">
            <Link to="/apply">Apply</Link>
          </Typography>
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
