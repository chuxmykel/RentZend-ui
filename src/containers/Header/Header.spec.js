import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

describe('<Header />', () => {
  const wrappedContainer = shallow(<Router><Header /></Router>);
  const container = shallow(<Header />);
  it('should match the snapshot', () => {
    expect(wrappedContainer.html()).toMatchSnapshot();
  });

  it('should have the main div', () => {
    expect(container.find('div').length).toEqual(1);
  });

  it('should have proper props for the main header div', () => {
    expect(container.find('div').prop('className')).toEqual('header');
    expect(container.find('div').prop('children')).toBeInstanceOf(Array);
  });

  it('should have 2 links to the different pages', () => {
    expect(container.find('Link').length).toEqual(2);
  });

  it('should have proper props for the links', () => {
    expect(container.find('Link').first().prop('to')).toEqual('/');
    expect(container.find('Link').last().prop('to')).toEqual('/apply');
  });
});
