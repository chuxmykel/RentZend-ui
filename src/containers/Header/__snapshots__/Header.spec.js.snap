// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div class=\\"header\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\"><a href=\\"/\\">View Agents</a></h6><h6 class=\\"MuiTypography-root MuiTypography-h6\\"><a href=\\"/apply\\">Apply</a></h6></div></div></header>"`;
